lib: location: Add LTE system mode sanity check (#7376)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

    The library is no longer directly referenced from the Application Event Manager.
    Instead, it uses the Application Event Manager hooks to connect with the manager.

* :ref:`lib_location`:

>   * Added sanity check to double check that the currently selected LTE system mode supports GNSS when using the GNSS location method for a location request.

Reviewer: ```suggestion
  * Added sanity check to verify that the currently selected LTE system mode supports GNSS when using the GNSS location method for a location request.
```

Reviewer 2: Nitpicking: I would replace "LTE" with "modem" because LTE spec does certainly not say anything about GNSS.

>     An error will be printed and the location request will return an error code if the sanity check fails.

Reviewer: ```suggestion
    An error is printed and the location request returns an error code if the sanity check fails.
```
